src/lighteval/metrics/dynamic_metrics.py

```diff
@@ -0,0 +1,71 @@
+# MIT License
+
+# Copyright (c) 2024 The HuggingFace Team
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+from typing import Callable
+
+import numpy as np
+
+from lighteval.metrics.metrics_sample import LoglikelihoodAcc, Probability
+from lighteval.metrics.normalizations import Normalization, PMINorm
+from lighteval.metrics.utils import MetricCategory, MetricUseCase, SampleLevelMetric
+
+
+def loglikelihood_acc_metric(normalization: Normalization | None = None) -> SampleLevelMetric:
```

Comment:
The be coherent with the rest of the codebase

```suggestion
def loglikelihood_acc_metric(normalization: Optional[Normalization] = None) -> SampleLevelMetric:
```

Reply:
will need to add the import statement

Reply:
OK so after looking it up, the prefered way is indeed using `type | None`, ^^'